### ISSUE TITLE
How were the default AVIF quality and effort settings chosen?

### ISSUE THREAD

**Issue by eeeps** [lovell/sharp#4227]
### What are you trying to achieve?

I am trying to decide which AVIF quality settings to use when using https://github.com/11ty/eleventy-img/, and found that tool's defaults (which rely on Sharp's defaults) [too low for my taste](https://github.com/11ty/eleventy-img/issues/198), for both quality and effort, in initial testing.

Before embarking on a bunch of tests, I'm looking for some context on how these scales work and how these values were chosen. For instance: is Sharp also just using libaom's defaults? (which are probably more appropriate for video than still images?).

### When you searched for similar issues, what did you find that might be related?

I found a few issues where [people are complaining about how expensive AVIF encoding is](https://github.com/lovell/sharp/issues/2597), and [some information about how Sharp's settings map to libaom's](https://github.com/lovell/sharp/issues/3418#issuecomment-1304421434), but nothing about how the scales work or how the defaults (which are both exactly in the middle of their range...) were chosen.

### Please provide sample image(s) that help explain this question

Here's the image that caused me to want to increase the defaults when using eleventy-img (all of the texture in the dog's face-fur vanishes, and in this usage context, I care much less about build times than compression performance):

Original: https://o.img.rodeo/w_800/dogs/2.png
Default settings (quality 50, effort 4): https://o.img.rodeo/w3fr2wrmmrf3pfrtmal7.avif – [compared to original](https://compare-images.glitch.me/?a_url=https%3A%2F%2Fo.img.rodeo%2Fw_800%2Fdogs%2F2.png&b_url=https%3A%2F%2Fo.img.rodeo%2Fw3fr2wrmmrf3pfrtmal7.avif&a_name=Original&b_name=quality%2050,%20effort%204&b_bytes=24384&a_format=PNG&b_format=AVIF)
Custom settings (quality 70, effort 7): https://o.img.rodeo/vahj96nhy8xchvlodlbo.avif – [compared to original](https://compare-images.glitch.me/?a_url=https%3A%2F%2Fo.img.rodeo%2Fw_800%2Fdogs%2F2.png&b_url=https%3A%2F%2Fo.img.rodeo%2Fvahj96nhy8xchvlodlbo.avif&a_name=Original&b_name=quality%2070,%20effort%207&b_bytes=47352&a_format=PNG&b_format=AVIF)

**Comment by eeeps** [comment]
Thank you for the prompt and super-helpful response!

Would it be accurate to say:
1. Sharp's effort and quality defaults were chosen because they are in the middle of libheif’s exposed range for each parameter,
2. those ranges map linearly to libaom's ranges, and
3. libaom's ranges were designed for video use cases.

Given the existing complaints about Sharp's slow/expensive AVIF encoding, and the wide range of use cases that Sharp is used for, it probably doesn't make sense to bump Sharp's default effort parameter. It might (after some more testing) make sense to bump it downstream in eleventy-image, though.

However – if my understanding above is correct – it might make sense to bump Sharp's AVIF quality default, because still images are always going to show their artifacts more than than video frames that flash by in an instant. Or maybe the (single) image I tested with is an outlier, and the AVIFs Sharp outputs by default are mostly fine for most use cases most of the time.

**Comment by lovell** [comment]
The default `quality` value of 50 was probably selected to match libheif's default `quality` value.

https://github.com/strukturag/libheif/blob/0a627a33d69cb5e93860536e841d23c8100f8065/libheif/plugins/encoder_aom.cc#L260

There was some research carried out a couple of years ago using sharp and dssim to calculate comparable "quality" settings for JPEG/WebP/AVIF.

https://www.industrialempathy.com/posts/avif-webp-quality-settings
https://github.com/cramforce/avif-webp-quality-setting

Interestingly, its summary that a JPEG quality of 80 is equivalent to an AVIF quality of 64 happens to align with the "visually lossless" suggested AVIF `cq-level` of 23 in the ffmpeg docs at https://trac.ffmpeg.org/wiki/Encode/AV1#ConstantQuality

Given sharp's default JPEG `quality` is currently 80, I could definitely be persuaded to default AVIF `quality` to a more comparable 64. Are you able to test with some of your sample images to see if quality=64 produces results that are subjectively closer to quality=70 than quality=50?

**Comment by eeeps** [comment]
> Are you able to test with some of your sample images to see if quality=64 produces results that are subjectively closer to quality=70 than quality=50?

Yes! But I'm about to leave for a week of vacation so not until I get back the week of October 14th.

**Comment by lovell** [comment]
@eeeps Were you able to make any progress with this?

**Comment by lovell** [comment]
@eeeps I'm considering a change in the default AVIF "quality" as part of the next release. Is there anything else from your research that you think might help?

**Comment by lovell** [comment]
If you hadn't seen, libaom v3.12.0 has added a new ssimulacra2-based tuning method named `iq` ("image quality").

https://aomedia.googlesource.com/aom/+/refs/tags/v3.12.0

I tried using this locally to compare `iq` against the default `ssim` tuning with a few sample images and observed that `iq` at a "quality" of 50 produces images that are around same file size as `ssim` will produce with a "quality" in the 60-65 range.

It might be the case here that we leave the default "quality" at 50 but change the default tuning to the newer `iq` where available, which I hope could produce perceptively better images at comparable files sizes (for a given `cq-level`).

**Comment by adityapatadia** [comment]
Why don't you expose the `tune` parameter so users can choose it? It might be helpful in different situations.

**Comment by eeeps** [comment]
@lovell This is just to say, 

1. Apologies for letting this slip through the cracks
2. ssimulacra2 is great (the author, Jon Sneyers is a colleague) and the AV1-SVT tuning for it also seems great. I was able to speak a couple of times to Gianni Rosato and Julio Barba as they developed it and it seems like they made some amazing improvements. Intuitively, and based on their tests, I would trust it to be more consistent and provide more performance than the ssim-tuning.

To check that with your own eyes, it sounds like a useful test right now would produce a couple dozen comparisons contrasting:

1. libaom ssim tuning vs iq tuning at sharp-quality 50
2. libaom ssim tuning vs iq tuning at sharp-quality 64

I'd probably use this set of lossless originals from Wikimedia to do it, although it's almost all photographic doesn't contain many synthetic images (screenshots, illustrations) which are also important. https://collection.cloudinary.com/eric-cloudinary/dedfc813424cc189902ce993d429f2db

**Comment by adityapatadia** [comment]
@lovell did we change the default tuning? If we want to change it, is there a way to achieve that?

**Comment by lovell** [comment]
v0.35.0 now available with `tune` option, defaults to `iq` for lossy.

https://sharp.pixelplumbing.com/api-output/#avif
